pkg/cmd/clients.go

```diff
@@ -130,68 +134,165 @@ func (cc *ClientCmd) CreateClientCmd() *cobra.Command {
 			}
+
 			name := args[0]
 			clientType := args[1]
 			appIdentifier := args[2]
-			appKey := uuid.NewV4().String()
-
 			namespace, err := currentNamespace(cmd.Flags())
 			if err != nil {
 				return errors.Wrap(err, "failed to get namespace")
 			}
-			app := &v1alpha1.MobileClient{
-				TypeMeta: metav1.TypeMeta{
-					Kind:       "MobileClient",
-					APIVersion: "mobile.k8s.io/v1alpha1",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Labels:      map[string]string{},
-					Annotations: map[string]string{},
-				},
-				Spec: v1alpha1.MobileClientSpec{Name: name, ApiKey: appKey, ClientType: clientType, AppIdentifier: appIdentifier},
+
+			clientId := name + "-" + clientType
+			client, _ := cc.mobileClient.MobileV1alpha1().MobileClients(namespace).Get(clientId, metav1.GetOptions{})
+			if client.ObjectMeta.UID != "" {
+				return errors.New("App with this name already exist for this client type")
 			}
-			switch app.Spec.ClientType {
+
+			var apbName string
+			switch clientType {
 			case "android":
-				app.Annotations["icon"] = "fa fa-android"
+				apbName = "android-app"
 				break
 			case "iOS":
-				app.Annotations["icon"] = "fa fa-apple"
+				apbName = "ios"
 				break
 			case "cordova":
-				app.Annotations["icon"] = "font-icon icon-cordova"
+				apbName = "cordova"
 				break
 			case "xamarin":
-				app.Annotations["icon"] = "font-icon icon-xamarin"
+				apbName = "xamarin"
 				break
-			default:
-				return errors.New("Unknown client type")
+
 			}
-			app.Name = name + "-" + strings.ToLower(app.Spec.ClientType)
-			if err := input.ValidateMobileClient(app); err != nil {
-				return errors.Wrap(err, "failed validation while creating new mobile client")
+			//Get available provision parameters from the cluster service plan
+			clusterServiceClass, err := findServiceClassByName(cc.scClient, apbName)
+			if err != nil {
+				return errors.WithStack(err)
 			}
-			createdClient, err := cc.mobileClient.MobileV1alpha1().MobileClients(namespace).Create(app)
+			clusterServicePlan, err := findServicePlanByNameAndClass(cc.scClient, "default", clusterServiceClass.Name)
 			if err != nil {
-				return errors.Wrap(err, "failed to create mobile client")
+				return errors.WithStack(err)
 			}
-			outType := outputType(cmd.Flags())
-			if err := cc.Out.Render("create"+cmd.Name(), outType, createdClient); err != nil {
-				return errors.Wrap(err, fmt.Sprintf(output.FailedToOutPutInFormat, "mobile client", outType))
+
+			instParams := &ServiceParams{}
+
+			if err := json.Unmarshal(clusterServicePlan.Spec.ServiceInstanceCreateParameterSchema.Raw, instParams); err != nil {
+				return errors.WithStack(err)
+			}
+
+			validServiceName := clusterServiceClass.Spec.ExternalName
+			extMeta := clusterServiceClass.Spec.ExternalMetadata.Raw
+			var extServiceClass ExternalServiceMetaData
+			if err := json.Unmarshal(extMeta, &extServiceClass); err != nil {
+				return errors.WithStack(err)
+			}
+
+			si := v1beta1.ServiceInstance{
```

Comment:
Do you think this might be better as a call to a constructor. I see similar creation in same pkg in at least one place https://github.com/aerogear/mobile-cli/blob/master/pkg/cmd/services.go#L231 have not checked for more but do you think it makes sense to move this out?